ember-server/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala

```diff
@@ -60,33 +65,12 @@ private[server] object ServerHelpers {
       additionalSocketOptions: List[SocketOptionMapping[_]] = List.empty,
       logger: Logger[F]
   )(implicit F: Concurrent[F], T: Timer[F]): Stream[F, Nothing] = {
-    def socketReadRequest(
-        socket: Socket[F],
-        requestHeaderReceiveTimeout: Duration,
-        receiveBufferSize: Int
-    ): F[Request[F]] = {
-      val idle: Option[FiniteDuration] = idleTimeout match {
-        case idle: FiniteDuration => Some(idle)
-        case _ => None
-      }
-      def errorIfEmpty(
-          socket: Socket[F],
-          maxBytes: Int,
-          timeout: Option[FiniteDuration]): Stream[F, Byte] =
-        Stream.eval(socket.read(maxBytes, timeout)).flatMap {
-          case Some(bytes) =>
-            Stream.chunk(bytes) ++ errorIfEmpty(socket, maxBytes, timeout)
-          case None => Stream.raiseError(new RuntimeException("Received Unexpected EOF"))
-        }
-      val action = requestHeaderReceiveTimeout match {
-        case i: FiniteDuration =>
-          Parser.Request.parser(maxHeaderSize, Some(i))(
-            errorIfEmpty(socket, receiveBufferSize, idle))
-        case _ =>
-          Parser.Request.parser(maxHeaderSize, None)(errorIfEmpty(socket, receiveBufferSize, idle))
+
+    def reachedEndError(socket: Socket[F]): Stream[F, Byte] =
+      Stream.eval(socket.read(receiveBufferSize, durationToFinite(idleTimeout))).flatMap {
+        case None => Stream.raiseError(new RuntimeException("Unexpected EOF"))
```

Comment:
Can we give this a more informative type than `RuntimeException`?

Reply:
Certainly. Custom or something else in mind?

Reply:
Actually, I think there's a standard one in `java.io` that would work great.